runtime/src/chpl-launcher-common.c

```diff
@@ -102,6 +103,42 @@ void chpl_append_to_largv(int* largc, const char*** largv, int* largv_len,
   (*largv)[(*largc)++] = (arg);
 }
 
+// Helper for appending arguments to a variable-size command buffer.
+// - Requires the buffer pointer is uninitialized on first call.
+// - After exceeding an initial allocated size estimate, each call will allocate
+//   additional memory as needed.
+char* chpl_append_to_cmd(char* cmdBuf, const char* format, ...) {
+  static const int initialSize = 2048;
+  static int charsWritten = 0;
+
+  va_list argsForLen, argsForPrint;
+  va_start(argsForLen, format);
+  va_copy(argsForPrint, argsForLen);
+
+  // Allocate buffer to initial size on first call
+  if (charsWritten == 0) {
+    assert(cmdBuf == NULL);
+    cmdBuf = (char*)chpl_mem_allocMany(initialSize, sizeof(char),
+                                    CHPL_RT_MD_COMMAND_BUFFER, -1, 0);
+  }
+
+  // Determine additional characters to be written
+  const int addedLen = vsnprintf(NULL, 0, format, argsForLen);
+  va_end(argsForLen);
+  int newLen = charsWritten + addedLen;
+
+  // Allocate more memory if needed
+  if (newLen >= initialSize) {
+    cmdBuf = (char*)chpl_mem_realloc(cmdBuf, newLen * sizeof(char),
+                                  CHPL_RT_MD_COMMAND_BUFFER, -1, 0);
+  }
```

Comment:
This approach works for me, however I have a concern. `chpl_mem_realloc` may invalidate the other pointers right? So if I call `chpl_append_to_cmd` and get a string in  the initial buffer, then call it again and get a string from a reallocated buffer. That could possibly invalidate the first string and leave a previous caller to this function with a dangling pointer, unless `cmdBuf` is modified by this function (its not, its passed by value)

Reply:
Yep, good catch, thanks. And I realized the newly-allocated memory doesn't even get the previous string copied into it... RAII has clearly dulled my instincts